napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+```
+napi_status napi_get_array_length(napi_env env, napi_value value, uint32_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: napi_value representing the JavaScript Array whose length is being queried
+- `[out] result`: uint32 representing length of the array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns the length of an array.
+
+Array length is described in [Section 22.1.4.1](https://tc39.github.io/ecma262/#sec-properties-of-array-instances-length) 
+of the ECMAScript Language Specification.
+
+### *napi_get_arraybuffer_info*
+
+#### Signature
+```
+napi_status napi_get_arraybuffer_info(napi_env env,
+                                      napi_value arraybuffer,
+                                      void** data,
+                                      size_t* byte_length)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  arraybuffer`: napi_value representing the ArrayBuffer being queried
+- `[out] data`: The underlying data buffer of the ArrayBuffer
+- `[out] byte_length`: Length in bytes of the underlying data buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to retrieve the underlying data buffer of an ArrayBuffer and it's length.
+WARNING: This API is dangerous. The lifetime of the underlying data buffer is managed by the 
+ArrayBuffer even after it's returned to you. The one possible safe to use this API is in conjunction 
+with napi_create_reference, which can be used to guarantee that you control the lifetime of
+the ArrayBuffer.
+
+<TODO: What happens to this buffer in case of a compacting GC?>
+
+### *napi_get_boolean*
+
+#### Signature
+```
+napi_status napi_get_boolean(napi_env env, bool value, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The value of the boolean to retrieve
+- `[out] result`: `napi_value` representing JavaScript Boolean singleton to retrieve
+
+#### Description
+This API is used to return the JavaScript singleton object that is used to represent the given boolean value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_buffer_info*
+
+#### Signature
+```
+napi_status napi_get_buffer_info(napi_env env,
+                                 napi_value value,
+                                 void** data,
+                                 size_t* length)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: napi_value representing the node::Buffer being queried
+- `[out] data`: The underlying data buffer of the node::Buffer
+- `[out] length`: Length in bytes of the underlying data buffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API is used to retrieve the underlying data buffer of an node::Buffer and it's length.
```

Comment:
This needs a similar warning about when/how long it is safe to use the data buffer.